### PR TITLE
use mirror.gcr.io instead of dockerhub images

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/priority important-soon
/sig testing

**What this PR does / why we need it**:

Dockerhub is going to rate-limit image pulls starting November 1st. In E2E tests, we are relying on a few images that are currently hosted on dockerhub. This can be problematic when we have so many presubmit test runs, leading to unrelated test failures.

One option would be to use ``mirror.gcr.io`` which mirrors some of the dockerhub images.

However, not all the images that we need can be found on ``mirror.gcr.io``: https://paste.ubuntu.com/p/5dmwtYP4CP/

The Redis image is no longer used; it was used for the guestbook application test, but that test has since been updated to use the agnhost image instead.

For more details, see: https://github.com/kubernetes/test-infra/issues/19477

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by mkumatag** [comment]
@claudiubelu will it fix the issue we are reported here - https://github.com/kubernetes/kubernetes/pull/95732?

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -67,6 +68,7 @@ func (i *Config) SetVersion(version string) {
 func initReg() RegistryList {
 	registry := RegistryList{
 		GcAuthenticatedRegistry: "gcr.io/authenticated-image-pulling",
+		DockerMirrorRegistry:    "docker.io/library",
```
Comment (dims) [inline test/utils/image/manifest.go:71]: shouldn't this be mirror.gcr.io?

Comment (claudiubelu) [reply test/utils/image/manifest.go:71]: oups, my bad. :)

**Comment by spiffxp** [comment]
I'm not sure we should be referencing mirror.gcr.io directly

**Comment by BenTheElder** [comment]
We can enable mirror.gcr.io in the container runtime configuration for CI everywhere (it should already be the case in most places) as a mirror, but we have no guarantee that these images will continue to be hosted at particular versions so I don't think we should reference it directly.

**Comment by spiffxp** [comment]
I think this is a no-op from a CI perspective
- prow build clusters have containerd configured to use mirror.gcr.io (ref: https://github.com/kubernetes/test-infra/issues/19477#issuecomment-714695917)
- clusters stood up with kube-up.sh use mirror.gcr.io as a mirror by default (ref: https://github.com/kubernetes/test-infra/issues/19477#issuecomment-714699818)

Might this help local developers?  Possibly, but I would also recommend going the route of having developers setup registry mirroring themselves, as they may be required to use something other than mirror.gcr.io

**Comment by mkumatag** [comment]
I agree with @spiffxp @BenTheElder on this because configuring the mirror in the container runtime will give us the flexibility to fall back to the docker hub if none of the mirrors cached that image and will also help us adding as many mirrors as we want.

I understand the initial pain in configuring the runtimes with the mirrors, but we can enhance the documentation with more information if required.

**Pull request closed without merging**